### PR TITLE
[Routing] Exposed "utf8" option, defaults "locale" and "format" in configuration

### PR DESCRIPTION
**Pull request by HeahDude**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ~
| License       | MIT
| Doc PR        | symfony/symfony-docs#11126

A sibling to #30501, everything is in the title :).

### CODE REVIEW

**Review by nicolas-grekas — DISMISSED** [review]

**Comment by HeahDude** [comment]
I've just added support for annotations too :). Doc is coming.

**Comment by fabpot** [comment]
Why is it useful to expose `compiler_class`?

**Comment by HeahDude** [comment]
Well, I was not sure, I did it by consistency. I felt like it was hidden and deserved to shine too :), but no hard feeling, if there is no real use case for it let's just keep utf8 which should increase DX (see related doc PR).

**Comment by fabpot** [comment]
I would remove it, that's an internal settings that does not need to be exposed.

**Review by fabpot — CHANGES_REQUESTED** [review]
`compiler_class`to be removed

**Comment by HeahDude** [comment]
Done, I exposed the defaults instead, WDYT?

Status: needs review

File: src/Symfony/Component/Routing/Annotation/Route.php [hunk 1]
```diff
@@ -53,6 +57,26 @@ public function __construct(array $data)
...
+            $data['defaults']['_locale'] = $data['locale'];
+            unset($data['locale']);
+        }
+
+        if (isset($data['format'])) {
+            $data['defaults']['_format'] = $data['format'];
+            unset($data['format']);
+        }
+
+        if (isset($data['fragment'])) {
```
Comment (fabpot) [inline src/Symfony/Component/Routing/Annotation/Route.php]: What's the use case?

Comment (HeahDude) [reply src/Symfony/Component/Routing/Annotation/Route.php]: I don't know if there is one yet, again I propose it for consistency, I can remove it :).

Comment (fabpot) [reply src/Symfony/Component/Routing/Annotation/Route.php]: We should only expose something if there is a use case. Let's remove everything that has no known use cases.

Comment (HeahDude) [reply src/Symfony/Component/Routing/Annotation/Route.php]: Ok done!

**Comment by fabpot** [comment]
@HeahDude Can you have a look at the broken tests?

**Review by nicolas-grekas — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @HeahDude.

**Pull request merged**